### ISSUE TITLE
telecentric calibration

### ISSUE THREAD

**Issue by mynickmynick** [opencv/opencv#25126]
### Describe the feature and motivation

dear all
please correct me if I am wrong: a telecentric lens model seems not to be covered in opencv calibration (as it is in Halcon mvTech for instance). I think it would be beneficial to implement it and add it. 

### Additional context

Please an opinion about it: in the meantime while it is not available, do you think it would be appropriate to get a telecentric model as a limit ( f-> infinite) just applying the projective lens pin hole model with a fixed "very high" focal (not recalculated) and then obtain the telecentric magnification m=f/z ? where z would be calculated by the calibration as a hopefully relatively high value as well? so basically the telecentric lens would be somehow simulated as a telephoto ? thank you very much in advance for your feedback

**Comment by AlexeyHaritonov** [comment]
Indeed needed feature

**Comment by mynickmynick** [comment]
regarding the (i quote myself) "Please an opinion about it: in the meantime while it is not available, do you think it would be appropriate to get a telecentric model as a limit ( f-> infinite) .." It seems to be not working with high initial fixrd f values. In the end I just used the ordinary calibration with f not fixed, whch is calculated with a relatively high value indeed (1.3076605184954961e+05=358mm in my case)

**Comment by AlexeyHaritonov** [comment]
mynickmynick  
did you find a way to calibrate telecentric lens?

**Comment by mynickmynick** [comment]
No, as I said I applied opencv projective model and i got back a veru high value of f , that could work just for the undistortion on the image plane but performed worse than mvtech telecentric model
